zhaquirks/thirdreality/plug.py

```diff
@@ -0,0 +1,53 @@
+"""Third Reality Plug devices."""
+
+from typing import Final
+
+from zigpy.quirks.v2 import ClusterType, add_to_registry_v2
+import zigpy.types as t
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+from zhaquirks import CustomCluster
+from zhaquirks.thirdreality import THIRD_REALITY
+
+
+class ThirdRealityPlugCluster(CustomCluster):
+    """The private cluster id of Third Reality's PLUG."""
+
+    cluster_id = 0xFF03
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Attribute of private cluster id."""
+
+        reset_summation_delivered: Final = ZCLAttributeDef(
+            id=0x0000,
+            name="reset_summation_delivered",
+            type=t.uint8_t,
+            is_manufacturer_specific=True,
+        )
+
+
+(
+    add_to_registry_v2(THIRD_REALITY, "3RSP02028BZ")
+    .adds(ThirdRealityPlugCluster)
+    .write_attr_button(
+        attribute_name=ThirdRealityPlugCluster.AttributeDefs.reset_summation_delivered.name,
+        attribute_value=0x01,
+        cluster_id=ThirdRealityPlugCluster.cluster_id,
+        cluster_type=ClusterType.Server,
+        endpoint_id=1,
+        translation_key="reset_summation_delivered"

```

Comment:
I think ZHA should already use the attribute name as the translation key by default. Since it's the same, this isn't needed IIRC.

Reply:
@TheJulianJES 
I'm sorry, I've been so busy lately, so I didn't pay attention to your reply in time.
I've made some changes to the code as you asked, and the modified code looks like this:

```
"""Third Reality Plug devices."""

from typing import Final

from zigpy.quirks import DeviceRegistry
from zigpy.quirks.v2 import ClusterType, add_to_registry_v2
import zigpy.types as t
from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef

from zhaquirks import CustomCluster
from zhaquirks.thirdreality import THIRD_REALITY


class ThirdRealityPlugCluster(CustomCluster):
    """The private cluster id of Third Reality's PLUG."""

    cluster_id = 0xFF03

    class AttributeDefs(BaseAttributeDefs):
        """Attribute of private cluster id."""

        reset_summation_delivered: Final = ZCLAttributeDef(
            id=0x0000,
            name="reset_summation_delivered",
            type=t.uint8_t,
            is_manufacturer_specific=True
        )


(
    add_to_registry_v2(THIRD_REALITY, "3RSP02028BZ")
    .also_applies_to(THIRD_REALITY, "3RSPE01044BZ")
    .adds(ThirdRealityPlugCluster.cluster_id)
    .replaces(ThirdRealityPlugCluster)
    .write_attr_button(
        attribute_name=ThirdRealityPlugCluster.AttributeDefs.reset_summation_delivered.name,
        attribute_value=0x01,
        cluster_id=ThirdRealityPlugCluster.cluster_id
    )
)
```
I'm not sure if the code changes are the same as you were asking for at the time, but the results don't work as expected. The name of this button in HA is still "Third Realtiy, Inc" + the model id of the device.
![image](https://github.com/zigpy/zha-device-handlers/assets/42085859/903e440f-dd95-40b5-9b30-c2ab08fb297e)
As shown in the following GIF, the button click function has been realized. So I don't know what is not enough to improve? Where do I need to modify? Can you please take some time out of your busy schedule to help me fix the imperfections.
![Operation Video](https://github.com/zigpy/zha-device-handlers/assets/42085859/830da11d-730f-4382-b103-fee11a7859ec)
There will be other devices that need to be supported in the future. I think if one device can realize the function, then there will be something to consider later.